### PR TITLE
The auction module and the bid service

### PR DESCRIPTION
**Pull request by AminaGhannem**

### CODE REVIEW

File: src/auction/bid.service.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  ) {
+    super(bidRepository);
+  }
+
+  getBidByPrice(price: number) {
+    this.bidRepository.find({ where: { price } });
+  }
+
+  async addBidToAuction(
+    bid: Partial<Bid>,
```
Comment (skandertebo) [inline src/auction/bid.service.ts:24]: the typing of argument bid is not correct, it should just be a type that has only a price and an auctionId, not an entire Partial<Bid>

File: src/auction/bid.service.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    private readonly auctionService: AuctionService,
+  ) {
+    super(bidRepository);
+  }
+
+  getBidByPrice(price: number) {
+    this.bidRepository.find({ where: { price } });
+  }
+
+  async addBidToAuction(
```
Comment (skandertebo) [inline src/auction/bid.service.ts:23]: you should check if the user has joined the auction

File: src/auction/auction.service.ts
Comment (skandertebo) [inline src/auction/auction.service.ts:1]: you should add a method that adds a user to an auction

File: src/auction/auction.resolver.ts [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  @Mutation(() => Auction)
+  createAuction(
+    @Args('createAuctionInput') createAuctionInput: CreateAuctionInput,
+    @GetUser() user: User,
+  ) {
+    const auction = {
+      ...createAuctionInput,
+      owner: user,
+      //5 hours from now
+      endTime: new Date(Date.now() + 5 * 60 * 60 * 1000).toISOString(),
```
Comment (MohamedRebai41) [inline src/auction/auction.resolver.ts:30]: Why is this hardcoded to 5 hours from now?

File: src/auction/entities/bid.entity.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+@ObjectType()
+@Entity()
+export class Bid extends GenericEntity {
+  @Field()
+  @Column()
+  price!: number;
+
+  @Field()
+  @CreateDateColumn()
```
Comment (MohamedRebai41) [inline src/auction/entities/bid.entity.ts:15]: You don't have to put this, it's already present in the generic entity

**Pull request merged**